### ISSUE TITLE
Single Table Inheritance with composite primary key not issuing correct SQL SELECT statement

### ISSUE THREAD

**Issue by aolvictim** [typeorm/typeorm#4252]
**Issue type:**

[ ] question
[X] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[X] `postgres`
[ ] `cockroachdb`
[X] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[ ] `latest`
[ ] `@next`
[X] `0.2.17` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**

Please take a look at the following code:

```typescript
// Person
@TableInheritance({ column: { type: "varchar", name: "type" } })
export class Person {

    @PrimaryGeneratedColumn('uuid')
    id: string;

    @PrimaryColumn()
    type: string; // type property is part of the composite primary key along with id

    @Column()
    name: string;
}

// User
@ChildEntity()
export class User extends Person { }

// create an user
let u1 = new User();
u1.name = "John Doe";
await connection.getRepository('User').save(u1);
```

I am using SQLite to test the code. Because SQLite does not return the inserted entry so `typeorm` issues an additional `SELECT` statement to retrieve the data just inserted. But look at the SQL log below you can see that the `SELECT` statement is incorrect.

```SQL
INSERT INTO "person"("id", "name", "type") VALUES (?, ?, ?) -- PARAMETERS: ["931f28ad-752d-48e2-b077-4c51755a535e","John Doe","User"] // this is correct

SELECT "User"."id" AS "User_id", "User"."type" AS "User_type" FROM "person" "User" WHERE  "User"."type" IN (?) -- PARAMETERS: ["User"] // this is NOT correct. "User"."id" = "931f28ad-752d-48e2-b077-4c51755a535e" should be part of the WHERE clause here
```

I dug a bit into the source code and found the problem is in the `insert` function in [ReturningResultsEntityUpdator.ts](https://github.com/typeorm/typeorm/blob/0.2.17/src/query-builder/ReturningResultsEntityUpdator.ts#L121):
```typescript
const entityId = metadata.getEntityIdMap(entity)!;
``` 
The `entityId` ends up being `undefined` while it suppose to be 
```javascript
{
    "id": "931f28ad-752d-48e2-b077-4c51755a535e",
    "type": "Person"
}
```

And the reason that's `undefined` is in `getEntityIdMap` function which calls `getValueMap` function in [EntityMetadata.ts](https://github.com/typeorm/typeorm/blob/0.2.17/src/metadata/EntityMetadata.ts#L749).

The `entity` passed in here is 
```javascript
{ name: 'John Doe', id: '7778b29e-9ba6-4a33-896d-f9f352886313' }
``` 
And `columns` contains the two primary columns in `Person`:
```javascript
[
    {id column},
    {type column}
]
```
Because the `entity` does not have the value for `type` column, so `getValueMap` returns `undefined` and this caused the `SELECT` query not having the correct `WHERE` clause.

I did an experiment by changing the code for `getValueMap` so instead of returning `undefined` when something is not found, it just ignores that property:
```typescript
// https://github.com/typeorm/typeorm/blob/0.2.17/src/metadata/EntityMetadata.ts#L755
return map; //undefined;
```
Then the correct SQL query is generated:
```SQL
 SELECT "User"."id" AS "User_id", "User"."type" AS "User_type" FROM "person" "User" WHERE ("User"."id" = ?) AND "User"."type" IN (?) -- PARAMETERS: ["98f05a23-6aea-46b6-8c60-0a4e712650d3","User"]
```

I am kinda stuck here and here are some thoughts:
- I am not sure if that's an actual fix since `getValueMap` is also used in `RelationMetadata.ts` and I don't know if it will cause other issues.
- Why doesn't the `entity` contain `type`? In the INSERT statement the `type` value is added automatically I assume because I am using a `User` repository so `type` does not need to be explicitly set. 
- Bottom line is primary columns for Person are `id` and `type`, but `getValueMap` cannot find value for `type` in the `entity` given.

Thanks

**Comment by aolvictim** [comment]
I came up with a potential fix. Here is the PR https://github.com/typeorm/typeorm/pull/4275
